uavcan/src/transport/can/mod.rs

```diff
@@ -1,336 +1,15 @@
-            return Err(RxError::NonLastUnderUtilization);
-        }
-
-        if CanServiceId(frame.id).is_svc() {
-            // Handle services
-            let id = CanServiceId(frame.id);
-
-            // Ignore invalid frames
-            if !id.valid() {
-                return Err(RxError::InvalidCanId);
-            }
-
-            // Ignore frames not meant for us
-            if node_id.is_none() || id.destination_id() != node_id.unwrap() {
-                return Ok(None);
-            }
-
-            let transfer_kind = if id.is_req() {
-                TransferKind::Request
-            } else {
-                TransferKind::Response
-            };
-
-            return Ok(Some(InternalRxFrame::as_service(
-                frame.timestamp,
-                Priority::from_u8(id.priority()).unwrap(),
-                transfer_kind,
-                id.service_id(),
-                id.source_id(),
-                id.destination_id(),
-                tail_byte.transfer_id(),
-                tail_byte.start_of_transfer(),
-                tail_byte.end_of_transfer(),
-                &frame.payload,
-            )));
-        } else {
-            // Handle messages
-            let id = CanMessageId(frame.id);
-
-            // We can ignore ID in anonymous transfers
-            let source_node_id = if id.is_anon() {
-                // Anonymous transfers can only be single-frame transfers
-                if !(tail_byte.start_of_transfer() && tail_byte.end_of_transfer()) {
-                    return Err(RxError::AnonNotSingleFrame);
-                }
-
-                None
-            } else {
-                Some(id.source_id())
-            };
-
-            if !id.valid() {
-                return Err(RxError::InvalidCanId);
-            }
-
-            return Ok(Some(InternalRxFrame::as_message(
-                frame.timestamp,
-                Priority::from_u8(id.priority()).unwrap(),
-                id.subject_id(),
-                source_node_id,
-                tail_byte.transfer_id(),
-                tail_byte.start_of_transfer(),
-                tail_byte.end_of_transfer(),
-                &frame.payload,
-            )));
-        }
-    }
-
-    fn transmit<'a>(
-        transfer: &'a crate::transfer::Transfer<C>,
-    ) -> Result<Self::FrameIter<'a>, TxError> {
-        CanIter::new(transfer, Some(1))
-    }
-}
-
-/// Iterator type to transmit a transfer.
-///
-/// By splitting transmission into an iterator I can easily `.collect()` it for a handy
-/// array, store it in another object, or just bulk transfer it all at once, without
-/// having to commit to any proper memory model.
-#[derive(Debug)]
-pub struct CanIter<'a, C: embedded_time::Clock> {
-    transfer: &'a crate::transfer::Transfer<'a, C>,
-    frame_id: u32,
-    payload_offset: usize,
-    crc: crc_any::CRCu16,
-    crc_left: u8,
-    toggle: bool,
-    is_start: bool,
-}
+pub use fd::*;
```

Comment:
I wouldn't remove the namespaces from the imports. I think importing the different structures like `transport::can::fd::CanFd` and `transport::can::legacy::Can` should be fine. Or do you think this is better if the two implementations will be merged in the future?